src/fastga_he/models/propulsion/components/loads/sm_pmsm/components/sizing_slot_height.py

```diff
@@ -0,0 +1,237 @@
+# This file is part of FAST-OAD_CS23-HE : A framework for rapid Overall Aircraft Design of Hybrid
+# Electric Aircraft.
+# Copyright (C) 2025 ISAE-SUPAERO
+
+import numpy as np
+import openmdao.api as om
+
+
+class SizingSlotHeight(om.ExplicitComponent):
+    """
+    Computation of single slot height of the SM PMSM in radial direction. The formula is obtained
+    from equation (II-46) in :cite:`touhami:2020`.
+    """
+
+    def initialize(self):
+        self.options.declare(
+            name="motor_id", default=None, desc="Identifier of the motor", allow_none=False
+        )
+
+    def setup(self):
+        motor_id = self.options["motor_id"]
+
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:"
+            + motor_id
+            + ":design_phase_current_density",
+            val=np.nan,
+            units="A/m**2",
+            desc="The design phase current density of each wire",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":winding_factor",
+            val=np.nan,
+            desc="The factor considers the cable winding effect in the stator slot",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":slot_conductor_factor",
+            val=np.nan,
+            desc="The area factor considers the cross-section shape twist due to wire bunching",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":slot_fill_factor",
+            val=np.nan,
+            desc="The factor describes the conductor material fullness inside the stator slots",
+        )
+        self.add_input(
+            name="data:propulsion:he_power_train:SM_PMSM:" + motor_id + ":tangential_stress",
```

Comment:
I'm only thinking about it now, but shouldn't the tangential stress depend on the rpm rating of the engine ?